src/tracepoint.jl

```diff
@@ -30,23 +30,55 @@ julia> x = rand(10,10);
 julia> @tracepoint "multiply" x * x;
 ```
 """
-macro tracepoint(name::String, ex::Expr)
-    return _tracepoint(name, ex, __module__, string(__source__.file), __source__.line)
+macro tracepoint(name::String, ex...)
+    kws, body = extract_keywords(ex)
+    return _tracepoint(name, body, __module__, string(__source__.file), __source__.line; kws...)
 end
 
-function _tracepoint(name::String, ex::Expr, mod::Module, filepath::String, line::Int)
+function _tracepoint(name::String, ex::Expr, mod::Module, filepath::String, line::Int; text=nothing, name2=nothing, color=nothing)
     srcloc = TracySrcLoc(name, nothing, filepath, line, 0, mod, true)
     push!(meta(mod), srcloc)
 
     N = length(meta(mod))
     m_id = getfield(mod, ID)
+
+    text_expr = text === nothing ? :() :
+        quote
+            local text = string($text)
+            @ccall libtracy.___tracy_emit_zone_text(ctx::TracyZoneContext,
+                                                    text::Cstring, length($text)::Csize_t)::Cvoid
+        end
+
+    name_expr = name2 === nothing ? :() :
+        quote
+            local name = $name2
+            if name isa Integer
+                @ccall libtracy.___tracy_emit_zone_value(ctx::TracyZoneContext,
```

Comment:
AFAIK value is more like the numerical analog to text than a zone name

My first-stroke thought is that in general the attributes that can be repeated multiple times per-zone (just text/value I think) should not be attached to the `@tracepoint`/`@zone` macro

We might even want to introduce an implicit "current zone" (in task-local storage?) for those functions, or the alternative is to provide an explicit `zone` object (which is valid to use only in the zone)